missing externals from libm

When I build NLopt, I get a lot of errors like:

```
nlopt/newuoa/newuoa.c:385: undefined reference to `sincos'
nlopt/isres/isres.c:93: undefined reference to `ceil'
nlopt/isres/isres.c:242: undefined reference to `exp'
nlopt/isres/isres.c:266: undefined reference to `sqrt'
nlopt/slsqp/slsqp.c:1325: undefined reference to `sqrt'
nlopt/esch/esch.c:39: undefined reference to `tan'
nlopt/api/optimize.c:458: undefined reference to `pow'
nlopt/util/mt19937ar.c:230: undefined reference to `log'
nlopt/util/mt19937ar.c:230: undefined reference to `sqrt'
nlopt/util/stop.c:199: undefined reference to `__fpclassify'
```

...and many more. Please add the missing link to libm.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,6 +72,7 @@ include (CheckTypeSize)
 include (CheckCCompilerFlag)
 include (CheckCXXSymbolExists)
 include (CheckCXXCompilerFlag)
+include (CheckLibraryExists)
 
 #==============================================================================
 # COMPILATION CHECKINGS and CONFIGURATION GENERATION
@@ -104,6 +105,11 @@ check_type_size ("uint32_t" SIZEOF_UINT32_T)
 set (HAVE_UINT32_T ${SIZEOF_UINT32_T})
 check_type_size ("unsigned int" SIZEOF_UNSIGNED_INT)
 check_type_size ("unsigned long" SIZEOF_UNSIGNED_LONG)
+
+check_library_exists ("m" sqrt "" HAVE_LIBM)
+if (HAVE_LIBM)
+  set (M_LIBRARY m)
+endif()
 
 if (NOT DEFINED HAVE_FPCLASSIFY)
   message(STATUS "Looking for fpclassify")
@@ -234,6 +240,8 @@ install (FILES ${NLOPT_HEADERS} DESTINATION ${INSTALL_INCLUDE_DIR})
 
 set (nlopt_lib nlopt${NLOPT_SUFFIX})
 add_library (${nlopt_lib} ${NLOPT_SOURCES})
+target_link_libraries (${nlopt_lib} ${M_LIBRARY})
+
 set_target_properties (${nlopt_lib} PROPERTIES SOVERSION 0)
 set_target_properties (${nlopt_lib} PROPERTIES VERSION 0.9.0)
 

